Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -344,8 +344,13 @@ void RenderBox::styleDidChange(StyleDifference diff, const RenderStyle* oldStyle
         if (auto* scrollableArea = layer()->scrollableArea())
             scrollableArea->scrollbarsController().scrollbarLayoutDirectionChanged(shouldPlaceVerticalScrollbarOnLeft() ? UserInterfaceLayoutDirection::RTL : UserInterfaceLayoutDirection::LTR);
     }
+
+    bool isDocElementRenderer = isDocumentElementRenderer();
+
     if (layer() && oldStyle && oldStyle->scrollbarWidth() != newStyle.scrollbarWidth()) {
-        if (auto* scrollableArea = layer()->scrollableArea())
+        if (isDocElementRenderer)
+            view().frameView().scrollbarsController().scrollbarWidthChanged(newStyle.scrollbarWidth());
+        else if (auto* scrollableArea = layer()->scrollableArea())
             scrollableArea->scrollbarsController().scrollbarWidthChanged(newStyle.scrollbarWidth());
```

Comment:
`ScrollableArea` has a virtual `scrollbarWidthChanged()` so I think it would be cleaner for the ScrollableArea to talk to its scrollbarsController.

Reply:
Fixed.